ldap/servers/slapd/back-ldbm/dbimpl.c

```diff
@@ -397,7 +397,44 @@ const char *dblayer_op2str(dbi_op_t op)
     return str[idx];
 }
 
-/* Open db env, db and db file privately */
+/* Get the li_directory directory from the database instance name -
+ * Caller should free the returned value
+ */
+static char *
+get_li_directory(const char *fname)
+{
+    /*
+     * li_directory is an existing directory.
+     * it can be fname or its parent or its greatparent
+     * in case of problem returns the provided name
+     */
+    char *lid = slapi_ch_strdup(fname);
+    struct stat sbuf = {0};
+    char *pt = NULL;
+    for (int i=0; i<3; i++) {
+        if (stat(lid, &sbuf) == 0) {
+            if (S_ISDIR(sbuf.st_mode)) {
+                return lid;
+            }
+            /* Non directory existing file could be regular
+             * at the first iteration otherwise it is an error.
+             */
+            if (i>0 || !S_ISREG(sbuf.st_mode)) {
+                break;
```

Comment:
I am missing how "break;" goes to error handling ?

Reply:
Originally there was a comment explaining that in error case the function returns a copy of the original string, since it is not a directory, there is a failure later on when trying to open the database
bit the comment get removed when I reorganized the code! :wink:
I will add it again. Anyway there iwill be a new commit because there s a regression about is_dbi_supported